python-avd/pyavd/_schema/models/avd_model.py

```diff
@@ -457,3 +458,59 @@ def _compare(self, other: object, ignore_fields: tuple[str, ...] = ()) -> bool:
 
         # Use regular __eq__ check on all nested models, since we do not carry over ignore_fields.
         return all(value == other.__dict__[field] for field, value in self.items() if field not in ignore_fields)
+
+    def _combine(self, other: Self) -> None:
+        """
+        Update instance by combining the other instance in.
+
+        Combining is different from merging in the sense that it will raise if there is a conflict
+        between one of our elements and the other elements.
+
+        for AvdIndexedList this is simply an append, if any conflict occurs a duplicate error will be raised.
+
+        Args:
+            other: The other instance of the same type to combine into this instance.
+
+        Raises:
+            AristaAvdDuplicateDataError: If any item from other is conflicting with an item from self when appending.
+        """
+        cls = type(self)
+        if not isinstance(other, cls):
+            msg = f"Unable to combine type '{type(other)}' into '{cls}'"
+            raise TypeError(msg)
+
+        # If the value of _created_from_null are different, consider it as a conflict.
+        if self._created_from_null != other._created_from_null:
+            raise AristaAvdDuplicateDataError(type(self).__name__, str(self._dump()), str(other._dump()))
+
+        for field, new_value in other.items():
+            old_value = self._get_defined_attr(field)
+
+            if old_value is Undefined:
+                # value not set so we can just pick the old_value
```

Comment:
Shouldn't be `new_value` here?

```suggestion
                # value not set so we can just pick the new_value
```